test/e2e/compatibility_test.go

```diff
@@ -105,3 +117,81 @@ query_tweaks:
 		return ret
 	}()
 }
+
+// nolint (it's still used in skipped test).
+type alwaysReadyProbe struct{}
+
+// nolint (it's still used in skipped test).
+func (p alwaysReadyProbe) Ready(e2e.Runnable) error { return nil }
+
+// TestAlertCompliance tests Alert compatibility against https://github.com/prometheus/compliance/blob/main/alert_generator.
+// NOTE: This requires dockerization of compliance framework: https://github.com/prometheus/compliance/pull/46
+func TestAlertCompliance_StatelessRuler(t *testing.T) {
+	t.Skip("This is an interactive test, using https://github.com/prometheus/compliance/tree/main/alert_generator testing is not optimized for CI runs")
+
+	e, err := e2e.NewDockerEnvironment("alert_compatibility")
+	testutil.Ok(t, err)
+	t.Cleanup(e.Close)
+
+	// Start receive + Querier.
+	receive, err := e2ethanos.NewIngestingReceiver(e, "receive")
```

Comment:
If we are using Thanos receiver in this case, the alerts will have the tenant label. During the alerts compliance check it will fail because it requires labels matching.